nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -225,8 +225,18 @@ in
               esac
             done
             ''] ++ (map (pool: ''
-            echo "importing root ZFS pool \"${pool}\"..."
-            zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}"
+            echo -n "importing root ZFS pool \"${pool}\"..."
+            trial=0
+            until msg=$(zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}" 2>&1); do
+              sleep 0.25
+              echo -n .
+              trial=$(($trial + 1))
+              if [[ $trial -eq 60 ]]; then
+                echo $msg
+                break
+              fi
+            done
+            echo
```

Comment:
I don't have any strong feelings about this either way. Here are some thoughts though:

1. The first 59 commands will have their output squashed
2. No back-off timer 
3. I'm not an expert on bash, so not 100% certain about the behavior of line 230

(in no way are these 3 thoughts a request to change it)